dlt/common/json/__init__.py

```diff
@@ -106,13 +107,13 @@ def custom_encode(obj: Any) -> str:
 # define decoder for each prefix
 DECODERS: List[Callable[[Any], Any]] = [
     Decimal,
-    parse_iso_like_datetime,
-    lambda s: parse_iso_like_datetime(s),
+    ensure_pendulum_datetime,
```

Comment:
we know the input is a string so here's no need to use `ensure_pendulum` this code is very time critical and if we can remove some code then better

Reply:
Yeah very true. Since we're only dealing with output from our own json encoder here I would just use the built in `fromisoformat`, that should be fastest.